src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -1347,6 +1348,46 @@ private static boolean loadAdvancements(Map<ResourceLocation, Advancement.Builde
         );
     }
 
+    private static final char[] ALPHABET = "abcdefghijklmnopqrstuvwxyz".toCharArray();
+    private static int[] lastCharIndices = new int[3];
+    /**For internal use only**/
+    public static String generateHorseArmorHash()
```

Comment:
Agreed. Resource locations ftw.

Reply:
It's part of the vanilla implementation. The hash is used to form a short unique ID for each combination of horse armour, base coat and markings. This ID is used to construct a ResourceLocation, which is used as the ID in the TextureManager for the LayeredTexture created from that combination. I'm not sure why Mojang has done this. Replacing the hash with the full texture location of the armour texture would greatly lengthen the ID, which would slow down its hashcode function and hence make the texture retrieval slower. However `String#hashCode()` is damn fast, run times are measured in nanoseconds. 

In the end I opted to keep the hash system, because Mojang has gone to some trouble to make it, so they must have a good reason. Making an automatic hash generator was easy, so I did that rather than removing the system and worrying about why Mojang made it.

Reply:
Ya, don't use small hashes. Its just super complex for no reason. I can ask around but im pretty sure the reason Mojang did it was because "Meh" Performance wise it isnt an issue.

Reply:
Okay, that simplifies things a lot.